code/modules/events/scrubber_overflow.dm

```diff
@@ -0,0 +1,112 @@
+/datum/round_event_control/scrubber_overflow
+	name = "Scrubber Overflow: Normal"
+	typepath = /datum/round_event/scrubber_overflow
+	weight = 10
+	max_occurrences = 3
+	min_players = 10
+
+/datum/round_event/scrubber_overflow
+	announce_when = 1
+	start_when = 5
+	end_when = 35
+	/// A list of scrubbers that will have reagents ejected from them
+	var/list/scrubbers = list()
+	/// The probability that the ejected reagents will be dangerous
+	var/danger_chance = 1
+	/// Amount of reagents ejected from each scrubber
+	var/reagents_amount = 50
+	/// The list of chems that scrubbers can produce
+	var/list/safer_chems = list(/datum/reagent/water,
+		/datum/reagent/carbon,
+		/datum/reagent/consumable/flour,
+		/datum/reagent/space_cleaner,
+		/datum/reagent/consumable/nutriment,
+		/datum/reagent/consumable/condensedcapsaicin,
+		/datum/reagent/drug/mushroomhallucinogen,
+		/datum/reagent/lube,
+		/datum/reagent/glitter/pink,
+		/datum/reagent/cryptobiolin,
+		/datum/reagent/toxin/plantbgone,
+		/datum/reagent/blood,
+		/datum/reagent/medicine/c2/multiver,
+		/datum/reagent/drug/space_drugs,
+		/datum/reagent/medicine/morphine,
+		/datum/reagent/water/holywater,
+		/datum/reagent/consumable/ethanol,
+		/datum/reagent/consumable/hot_coco,
+		/datum/reagent/toxin/acid,
+		/datum/reagent/toxin/mindbreaker,
+		/datum/reagent/toxin/rotatium,
+		/datum/reagent/bluespace,
+		/datum/reagent/pax,
+		/datum/reagent/consumable/laughter,
+		/datum/reagent/concentrated_barbers_aid,
+		/datum/reagent/baldium,
+		/datum/reagent/colorful_reagent,
+		/datum/reagent/peaceborg/confuse,
+		/datum/reagent/peaceborg/tire,
+		/datum/reagent/consumable/salt,
+		/datum/reagent/consumable/ethanol/beer,
+		/datum/reagent/hair_dye,
+		/datum/reagent/consumable/sugar,
+		/datum/reagent/glitter/white,
+		/datum/reagent/growthserum,
+	)
+	//needs to be chemid unit checked at some point
+
+/datum/round_event/scrubber_overflow/announce()
+	priority_announce("The scrubbers network is experiencing a backpressure surge. Some ejection of contents may occur.", "Atmospherics alert")
+
+/datum/round_event/scrubber_overflow/setup()
+	end_when = rand(25, 100)
+	for(var/obj/machinery/atmospherics/components/unary/vent_scrubber/temp_vent in GLOB.machines)
+		var/turf/scrubber_turf = get_turf(temp_vent)
+		if(scrubber_turf && is_station_level(scrubber_turf.z) && !temp_vent.welded)
+			scrubbers += temp_vent
```

Comment:
how much time does this take?

Reply:
According to a test on Deltastation, 
```
[2022-08-23 04:17:09.409] GAME: Starting to find scrubbers took 0 deciseconds
[2022-08-23 04:17:09.419] GAME: Found 292 scrubbers took 0.09375 deciseconds
```
```
[2022-08-23 04:20:32.367] GAME: Starting to find scrubbers took 0 deciseconds
[2022-08-23 04:20:32.376] GAME: Found 287 scrubbers took 0.09375 deciseconds
```
```
[2022-08-23 04:21:07.324] GAME: Starting to find scrubbers took 0 deciseconds
[2022-08-23 04:21:07.334] GAME: Found 287 scrubbers took 0.09375 deciseconds
```

(The acid burned away some scrubbers so that's why we lost a couple)

Reply:
The code for checking was basically the same as in your initialization time checking logging